frontend/packages/ceph-storage-plugin/integration-tests/tests/add-capacity.scenario.ts

```diff
@@ -0,0 +1,173 @@
+  ocsOp,
+  storageClusterRow,
+  storageClusterView,
+  verifyFields,
+} from '../views/add-capacity.view';
+import { CLUSTER_STATUS, EXPAND_WAIT, KIND, NS, POD_NAME_PATTERNS, SECOND } from '../utils/consts';
+import {
+  getOSDPreparePodsCnt,
+  getPodName,
+  getPodPhase,
+  getPodRestartCount,
+  isPodPresent,
+  testPodIsRunning,
+  testPodIsSucceeded,
+} from '../utils/helpers';
+
+const storageCluster = JSON.parse(execSync(`kubectl get -o json -n ${NS} ${KIND}`).toString());
+const cephValue = JSON.parse(execSync(`kubectl get cephCluster -n ${NS} -o json`).toString());
+const clusterStatus = storageCluster.items[0];
+const cephHealth = cephValue.items[0];
+
+const expansionObjects = {
+  clusterJSON: {},
+  previousCnt: 0,
+  updatedCnt: 0,
+  updatedClusterJSON: {},
+  previousPods: { items: [] as PodKind[] },
+  updatedPods: { items: [] as PodKind[] },
+};
+
+describe('Check availability of ocs cluster', () => {
+  if (clusterStatus) {
+    it('Should check if the ocs cluster is Ready for expansion', () => {
+      expect(_.get(clusterStatus, 'status.phase')).toBe(CLUSTER_STATUS.READY);
+    });
+  } else {
+    it('Should state that ocs cluster is not ready for expansion', () => {
+      expect(clusterStatus).toBeUndefined();
+    });
+  }
+});
+
+describe('Check availability of Ceph cluster', () => {
+  if (cephHealth) {
+    it('Check if the Ceph cluster is healthy before expansion', () => {
+      expect(cephHealth.status.ceph.health).not.toBe(CLUSTER_STATUS.HEALTH_ERROR);
+    });
+  } else {
+    it('Should state that Ceph cluster doesnt exist', () => {
+      expect(cephHealth).toBeUndefined();
+    });
+  }
+});
+
+if (clusterStatus && cephHealth) {
+  describe('Check add capacity functionality for ocs service', () => {
+    beforeAll(async () => {
+      [expansionObjects.clusterJSON] = storageCluster.items;
+      const name = getName(expansionObjects.clusterJSON);
+      expansionObjects.previousCnt = _.get(
+        expansionObjects.clusterJSON,
+        'spec.storageDeviceSets[0].count',
+      );
+      const uid = _.get(expansionObjects.clusterJSON, 'metadata.uid').toString();
+      expansionObjects.previousPods = JSON.parse(
+        execSync(`kubectl get pods -n ${NS} -o json`).toString(),
+      );
+
+      await goToInstalledOperators();
+      await click(ocsOp);
+      await click(storageClusterView);
+
+      await clickKebabAction(uid, 'Add Capacity');
+      await verifyFields();
+      await click(confirmButton);
+
+      const statusCol = storageClusterRow(uid).$('td:nth-child(4)');
+
+      // need to wait as cluster states fluctuates for some time. Waiting for 2 secs for the same
+      await browser.sleep(2 * SECOND);
+
+      await browser.wait(until.textToBePresentInElement(statusCol, CLUSTER_STATUS.PROGRESSING));
+      await browser.wait(
+        until.textToBePresentInElement(
+          statusCol.$('span.co-icon-and-text span'),
+          CLUSTER_STATUS.READY,
+        ),
+      );
```

Comment:
Please add a comment or put this as a function with the name waiting.

Reply:
Expect block for ready state too.

Reply:
Not a blocker thing. Can be addressed in the follow up PR. We need to get this PR merge soon.